### PR TITLE
Added support for querying user by email #6528

### PR DESCRIPTION
**Pull request by LeOndaz**
Note that if the user supplied both ID and email, it will work (should I change this?)
Some tests don't pass on my machine (even without my edits), this is why I'll be testing here

### CODE REVIEW

File: saleor/graphql/account/schema.py [hunk 1]
```diff
@@ -169,8 +172,8 @@ def resolve_staff_users(self, info, query=None, **kwargs):
     @one_of_permissions_required(
         [AccountPermissions.MANAGE_STAFF, AccountPermissions.MANAGE_USERS]
     )
-    def resolve_user(self, info, id):
-        return resolve_user(info, id)
+    def resolve_user(self, info, id=None, email=None):
+        return resolve_user(info, id, email)
```
Comment (maarcingebala) [inline saleor/graphql/account/schema.py:176]: We have a `validate_one_of_args_is_in_query` validator which is used in queries like this to check if any of the two params was passed, as it is required to pass one of them.

File: saleor/graphql/account/schema.py [hunk 2]
```diff
@@ -131,8 +131,11 @@ class AccountQueries(graphene.ObjectType):
     )
     user = graphene.Field(
         User,
-        id=graphene.Argument(graphene.ID, description="ID of the user.", required=True),
-        description="Look up a user by ID.",
+        id=graphene.Argument(graphene.ID, description="ID of the user."),
+        email=graphene.Argument(
+            graphene.String, description="Email address of the user."
+        ),
+        description="Look up a user by email address.",
```
Comment (maarcingebala) [inline saleor/graphql/account/schema.py:138]: "Look up a user by ID or email address."

File: saleor/graphql/account/resolvers.py [hunk 3]
```diff
@@ -50,18 +50,22 @@ def resolve_staff_users(info, query, **_kwargs):
     return qs.distinct()
 
 
-def resolve_user(info, id):
+def resolve_user(info, id=None, email=None):
     requester = get_user_or_app_from_context(info.context)
     if requester:
-        _model, user_pk = graphene.Node.from_global_id(id)
+        filterkwargs = {}
```
Comment (maarcingebala) [inline saleor/graphql/account/resolvers.py:56]: In Python it should be `filter_kwargs`

File: saleor/graphql/account/resolvers.py [hunk 4]
```diff
@@ -50,18 +50,22 @@ def resolve_staff_users(info, query, **_kwargs):
     return qs.distinct()
 
 
-def resolve_user(info, id):
+def resolve_user(info, id=None, email=None):
```
[the remark this answers is not in the document]
Comment (fowczarek) [reply saleor/graphql/account/resolvers.py:53]: We should add a test for the situation when 2 args are passed in the query.

File: saleor/graphql/account/schema.py (hunk 1 above)
Comment (LeOndaz) [reply saleor/graphql/account/schema.py:176]: Yep thanks, I actually did the same in attributes but tbh, I totally forgot about the existence of this PR, welcome back! I'll fix everything then commit.

**Comment by LeOndaz** [comment]
okay, the last commit is just reformatted with black & isort as I forgot, this is working, only 1 parameter can be passed.

File: saleor/graphql/account/resolvers.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (LeOndaz) [reply saleor/graphql/account/resolvers.py:53]: @fowczarek okay, I'll add that later

[the remark this answers is not in the document]
Comment (LeOndaz) [reply saleor/graphql/account/resolvers.py:53]: @fowczarek okay I'll do it

File: saleor/graphql/account/resolvers.py (hunk 3 above)
Comment (LeOndaz) [reply saleor/graphql/account/resolvers.py]: I actually named it like that inspired by how django calls it "initkwargs" and not "init_kwargs", but I've done what you said in the commit anyway

File: saleor/graphql/account/resolvers.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (LeOndaz) [reply saleor/graphql/account/resolvers.py:53]: @fowczarek I've added the test

**Comment by LeOndaz** [comment]
mb, this should be the latest commit

**Review by fowczarek — APPROVED** [review]

**Review by IKarbowiak — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

**Comment by maarcingebala** [comment]
@LeOndaz Please add a changelog entry to the bottom of the "Unreleased" section.

**Comment by LeOndaz** [comment]
> @LeOndaz Please add a changelog entry to the bottom of the "Unreleased" section.



> @LeOndaz Please add a changelog entry to the bottom of the "Unreleased" section.

Okay I've done that

**Comment by LeOndaz** [comment]
Okay finally, I didn't notice that someone committed something today this is why the test was failing, hopefully it should be fine now

**Pull request merged**